3rdparty/ngraph_bridge/src/ngraph_subgraph.h

```diff
@@ -27,39 +28,54 @@ using namespace mxnet::op;
 
 class SgNgraphSelector : public SubgraphSelector {
  public:
-  SgNgraphSelector() {}
+  SgNgraphSelector(const nnvm::Graph &g) : compiler_(g) {}
 
   bool Select(const nnvm::Node &n) override {
-    bool match = !n.is_variable() && (n.attrs.name.substr(0, 6) == "ngraph");
-    return match;
+    return (!n.is_variable() && is_node_selected(n));
   }
 
   bool SelectInput(const nnvm::Node &n, const nnvm::Node &new_node) override {
-    return false;
+    return (!n.is_variable() && is_node_selected(new_node));
   }
 
   bool SelectOutput(const nnvm::Node &n, const nnvm::Node &new_node) override {
+    return (!n.is_variable() && is_node_selected(new_node));
+  }
+
+ private:
+  Compiler compiler_;
+  bool is_node_selected(const nnvm::Node &n) {
+    NodePtr nn;
+    MapEntry tmp{&n, 0};
```

Comment:
Oh, I see, the subgraph API is forcing const ref, you don't have a shared pointer.